File: app/src/main/java/io/ipoli/android/quest/data/Tag.java
package io.ipoli.android.quest.data;

import java.util.Date;

import io.ipoli.android.app.persistence.PersistedObject;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 3/26/16.
 */
public class Tag extends PersistedObject {

    private String name;

    public Tag() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean getIsDeleted() {
        return isDeleted;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }
}

File: app/src/main/java/io/ipoli/android/tag/Tag.kt
package io.ipoli.android.tag

import io.ipoli.android.quest.Color
import io.ipoli.android.quest.Entity
import io.ipoli.android.quest.Icon
import org.threeten.bp.Instant

/**
 * Created by Venelin Valkov <venelin@mypoli.fun>
 * on 04/03/2018.
 */
data class Tag(
    override val id: String = "",
    val name: String,
    val color: Color,
    val icon: Icon? = null,
    val isFavorite: Boolean,
    val questCount: Int = -1,
    override val createdAt: Instant = Instant.now(),
    override val updatedAt: Instant = Instant.now()
) : Entity